apps/frontend/src/app/CreateOrder/CreateOrder.tsx

```diff
@@ -1,51 +1,115 @@
-import React, { FC, useState } from 'react';
+import React, { ChangeEvent, FC, useState } from 'react';
 import { useForm } from 'react-hook-form';
 import { useHistory } from 'react-router-dom';
 
 import type { CreateOrderDto, OrderModel } from '@pasnik/api/data-transfer';
 import { useOrdersFacade } from '@pasnik/orders-data-access';
+import { Alert, Button, InputAdornment, Paper, TextField } from '@mui/material';
+import { formatter, validUrl } from '../../utils';
+
 export const CreateOrder: FC = () => {
   const { register, handleSubmit } = useForm();
   const { createOrder } = useOrdersFacade();
 
+  const [restaurant, setRestaurant] = useState<string>('');
+  const [menu, setMenu] = useState<string>('');
+  const [isValid, setIsValid] = useState<boolean>(false);
+  const [isValidInput, setIsValidInput] = useState<boolean>(false);
+  const [DeliveryPrice, setDeliveryPrice] = useState<string>('');
+
   const [error, setError] = useState<string | null>(null);
   const history = useHistory();
 
   const onSubmit = (data: CreateOrderDto) => {
-    createOrder({ ...data, orderAt: new Date().toISOString() })
-      .then((params: OrderModel) => history.push(`/`))
+    const newPrice = data.shippingCents
+      ? +formatter.format(data.shippingCents)
+      : -1;
+    createOrder({
+      ...data,
+      orderAt: new Date().toISOString(),
+      shippingCents: newPrice * 100,
+    })
+      .then((params: OrderModel) => {
+        history.push(`order/` + params.id);
+      })
       .catch((err: Error) => setError(err.message));
   };
 
-  const clearError = () => {
-    setError(null);
-  };
+  function handleChangeRestaurant(e: ChangeEvent<HTMLInputElement>) {
+    setRestaurant(e.currentTarget.value);
+  }
+
+  function handleChangeMenu(e: ChangeEvent<HTMLInputElement>) {
+    setMenu(e.currentTarget.value);
+    setIsValid(validUrl(e.currentTarget.value));
+    setIsValidInput(validUrl(e.currentTarget.value));
+  }
+
+  function handleChangeDeliveryPrice(e: ChangeEvent<HTMLInputElement>) {
+    setDeliveryPrice(e.currentTarget.value);
+  }
```

Comment:
@lukszymanski do you think those should be `const`s and wrapped with `useCallback` or something? 🤔